### ISSUE TITLE
Unable to compile dynamic module with thumbnail and encryption functionalities

### ISSUE THREAD

**Issue by nixklai** [kaltura/nginx-vod-module#1152]
I recently try compiling nginx-vod-module on ubuntu 18.04 and nginx 1.18.0 as a dynamic module; however, the thumbnail functionality is disabled due to the lack of `libavcodec` and such. Also, the directives `vod_hls_encryption_method` and `vod_hls_encryption_key_file_name` are not recognized.

As suggested by other comments, I try compiling `FFmpeg` myself to get around it, but I still can't fix the bugs that stop me from compiling.

I have summarised many of the build instruction into two repositories, [FFmpeg prebuilt](https://github.com/nixklai/ffmpeg-shared-prebuilt) and [nginx-vod-module prebuilt](https://github.com/nixklai/nginx-vod-module-prebuilt). Please generously offer your advice. Thanks.

**Comment by jessp01** [comment]
Hello @nixklai ,

My guess would be that you do not have the OpenSSL headers on the machine you compiled the module on. These are needed for encryption support.
Further note that some functions also require FFmpeg to be compiled with FDK support.

**Comment by jessp01** [comment]
Hi @nixklai,

Any news on this?

**Comment by nixklai** [comment]
Hi @jessp01,

I am still kinda noob on compiling modules, and I don't know much what I am doing. 😢 
I will continue to try my best, but I think I am stuck on compiling FFMPEG with the necessary dependency.
I try not to compile FFMPEG since I want [my package](https://github.com/nixklai/nginx-vod-module-prebuilt) to be a kinda turn-key solution, but I am always missing some libraries here and there when compiling FFMPEG myself.

**TL;DR** I need some more help on getting (preferably not compiling) the required FFMPEG-related libraries.
Again, many thanks!
